faker/providers/internet/__init__.py

```diff
@@ -169,20 +170,56 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self, network=False):
-        """Produce a random IPv4 address or network with a valid CIDR"""
-        address = str(ip_address(self.generator.random.randint(
-            0, (2 ** IPV4LENGTH) - 1)))
-        if network:
+    def ipv4(self, network=False, private=False, simple=True):
```

Comment:
I'm -1 on `simple`. Users should not be expected to worry about how `Faker` generates the value.

Reply:
Yes -1 on simple